.github/workflows/ios-build-and-test.yml

```diff
@@ -0,0 +1,139 @@
+name: Build and Test (iOS)
+
+on:
+  workflow_call:
+    inputs:
+      ios-device-names:
+        description: |
+          The iOS device names to use for testing. Must be formatted as a valid JSON array string.
+          Example: '["iPhone 15", "iPhone 15 Pro"]'
+            - Notice the use of single quotes to wrap the JSON array string, and double quotes for each array element.
+        type: string
+        default: '["iPhone 15"]'
+      ios-versions:
+        description: |
+          The iOS versions to use for testing. Must be formatted as a valid JSON array string.
+          Example: '["18.0", "18.1"]'
+            - Notice the use of single quotes to wrap the JSON array string, and double quotes for each array element.
+        type: string
+        default: '["18.0"]'
+      tvos-device-names:
+        description: |
+          The tvOS device names to use for testing. Must be formatted as a valid JSON array string.
+          Example: '["Apple TV 4K (3rd generation)", "Apple TV"]'
+            - Notice the use of single quotes to wrap the JSON array string, and double quotes for each array element.
+        type: string
+        default: '["Apple TV"]'
+      tvos-versions:
+        description: |
+          The tvOS versions to use for testing. Must be formatted as a valid JSON array string.
+          Example: '["18.0", "18.1"]'
+            - Notice the use of single quotes to wrap the JSON array string, and double quotes for each array element.
+        type: string
+        default: '["18.0"]'
+      # Flags for which tests to run
+      run-test-ios-unit:
+        type: boolean
+        default: false
+      run-test-ios-functional:
+        type: boolean
+        default: false
+      run-test-ios-integration:
+        type: boolean
+        default: false
+      run-test-tvos-unit:
+        type: boolean
+        default: false
+      run-test-tvos-functional:
+        type: boolean
+        default: false
+      run-test-tvos-integration:
+        type: boolean
+        default: false
+      run-build-xcframework-and-app:
+        type: boolean
+        default: false
+
+jobs:
+  validate-code:
+    name: Validate Code
+    uses: timkimadobe/aepsdk-commons/.github/workflows/ios-validate-code.yml@ios-circleci-to-github
+  
+  test-ios-unit:
+    name: Unit Test (iOS)
+    needs: validate-code
+    if: inputs.run-test-ios-unit
+    uses: timkimadobe/aepsdk-commons/.github/workflows/ios-custom-command-build-and-test.yml@ios-circleci-to-github
+    with:
+      ios-device-names: ${{ inputs.ios-device-names }}
+      ios-versions: ${{ inputs.ios-versions }}
+      command: make unit-test-ios
```

Comment:
Can you also add just one flag to enable codecov (with default=true) for the entire workflow and add necessary steps for uploading the coverage reports?